ExcelWriter.py

```diff
@@ -19,6 +20,36 @@ def __init__(self, name):
             self.maxCol = []
             self.indentFormat = {}
             self.name=name
+            self.writtenCells = {}
+            
+    def Add_Header(self, headerLines):
+        row = 0;
+        maxCol = 0;
+        for header in headerLines:
+            col = 0
+            for headerLine in header:
+                self.Write_String(row, col, headerLine)
+                col = col + 1
+                maxCol = max(col, maxCol)
+        self.currentCol = maxCol
+        
+    
+    def Write_String(self, row, col, stringVal, formatValue):
+        # by putting chr(2) at the front of the string, google docs will not remove
+        # the leading spaces
+        tabString = " " * 4
+        stringVal = stringVal.replace("\t", tabString)
+        self.Inc_Max_Col(stringVal)
+        
+
+        stringVal = chr(2) + stringVal;
```

Comment:
Is this prepending an STX byte? I don't follow what the purpose of this line is.

Reply:
Yes, by prepending the STX, google sheets will no longer filter out the leading spaces, so it preserves the formatting of the original output/file (This is actually the main purpose of this commit.    The rest is stuff that is only halfway done, but I wanted to get that change in there).

Reply:
Wow, how did you figure this out?

Reply:
Google sheets basically just does a trim.   So, I wanted something in there that was invisible, but prevented the removing of the leading spaces.